libcnb/src/build.rs

```diff
@@ -27,6 +19,21 @@ pub struct BuildContext<B: Buildpack + ?Sized> {
     pub buildpack_descriptor: BuildpackToml<B::Metadata>,
 }
 
+impl<B: Buildpack + ?Sized> BuildContext<B> {
+    pub fn handle_layer<L: Layer<Buildpack = B>>(
+        &self,
+        name: impl AsRef<str>,
+        layer: L,
+    ) -> crate::Result<LayerData<L::Metadata>, B::Error> {
+        crate::layer::handle_layer(self, name.as_ref(), layer).map_err(|error| match error {
+            HandleLayerErrorOrBuildpackError::HandleLayerError(e) => {
+                crate::Error::HandleLayerError(e)
+            }
+            HandleLayerErrorOrBuildpackError::BuildpackError(e) => crate::Error::BuildpackError(e),
```

Comment:
> Maybe we rename public to just "layer.rs"?

This would result in this module structure: `crate::layer::layer::Layer` which is a bit weird, isn't it? Given that we only expose a single `layer` module to users to our users, this is all internal though. If you feel strongly about this we have to re-think all of the modules in libcnb since we use quite a bunch of `pub use` to carefully craft what we expose and under which module structure.